lib/OpenQA/Schema/Result/Workers.pm

```diff
@@ -87,8 +88,15 @@ sub update_caps {
 sub get_property {
     my ($self, $key) = @_;
 
-    my $r = $self->properties->find({key => $key});
-    return $r ? $r->value : undef;
+    # Optimized because this is a performance hot spot for the websocket server
+    my $sth = $self->result_source->schema->storage->dbh->prepare(
```

Comment:
Can you try out if the following optimization would be enough?
```
my $r = $self->properties->find({key => $key}, {select => [qw(value)]});
```

Reply:
One of the first things i tried, makes literally no difference at all, still `1.34s`.

Reply:
Before you ask, I also tried `->search()` with explicit `LIMIT 1`. Also no difference, it's all in the DBIx::Class overhead.

Reply:
The `LIMIT 1` doesn't do much work here btw., i only have it in the raw SQL because it's the correct query to use. :grin:

Reply:
Here's all the numbers, since you asked for that on Slack:
<img width="1851" src="https://github.com/os-autoinst/openQA/assets/30094/41e2bf71-10dc-49ba-8a0c-331665be6183">

Reply:
So it spends about as much time on the `->proterties` call, as the whole raw SQL version takes performing queries.

Reply:
Ok.
with DBIx::Class overhead I also associate overhead for creating DateTime objects for columns that are often never used.
I just made a small comparison with Time::Hires, and I also don't really see a big difference.
Maybe DBIx::Class can now lazy-load columns like this?

It wasn't clear to me that you had checked this, so I had to ask.

Reply:
Never underestimate the cost of plain old object instantiation in Perl when it comes to performance hot spots.